.buildkite/dockerfiles-pipeline.yml

```diff
@@ -0,0 +1,124 @@
+definitions:
+  steps:
+    - step: &test-agents
+        agents:
+          provider: "gcp"
+          machineType: "n1-standard-8"
+          useVault: true
+          image: family/enterprise-search-ubuntu-2204-connectors-py
+  retries: &retries
+    retry:
+      automatic:
+        - exit_status: -1  # Connection to the Agent was lost
+          signal_reason: none
+          limit: 2
+        - exit_status: 255  # Timeout
+          signal_reason: none
+          limit: 2
+        - exit_status: 2  # Flaky test
+          signal_reason: none
+          limit: 2
+
+steps:
+  - group: ":truck: Building, Testing and Scanning Dockerfile and Dockerfile.ftest"
+    key: "build_test_scan_group"
+    if: "(build.branch == \"main\")"
+    steps:
+      # ----
+      # Dockerfile build and tests on amd64
+      # ----
+      - label: "Build amd64 image from Dockerfile"
```

Comment:
instead of only building `amd64` images, we could consider building a multi-arch image using `buildah` or `drivah`, like [we are doing here](https://github.com/elastic/connectors/blob/main/.buildkite/release-pipeline.yml#L142-L150), and that way we could also potentially avoid using these hefty ec2 instances

Reply:
Thank you for the suggestion about a multi-arch image, reading in more detail how buildah/drivah [multiarch push](https://github.com/elastic/connectors/blob/main/.buildkite/release-pipeline.yml#L142C17-L142C54) works, it [pulls](https://github.com/elastic/connectors/blob/main/.buildkite/publish/build-multiarch-docker.sh#L20) the already built amd64/arm64 images from the previous steps and combines them into a common manifest which references both builds. Unfortunately doesn't save us any build resources. 

Still, a good to have - but it means that we'd build the image on both architectures. Let me know of your thoughts.